trojsten/polls/templatetags/polls_parts.py

```diff
@@ -10,54 +11,16 @@
 def show_time(context, current_question):
     start = current_question.created_date
     end = current_question.deadline
-    full = end - start
-    remaining = end - timezone.now()
-    elapsed = full - remaining
-    if remaining.days <= settings.ROUND_PROGRESS_DANGER_DAYS:
-        progressbar_class = settings.ROUND_PROGRESS_DANGER_CLASS
-    elif remaining.days <= settings.ROUND_PROGRESS_WARNING_DAYS:
-        progressbar_class = settings.ROUND_PROGRESS_WARNING_CLASS
-    else:
-        progressbar_class = settings.ROUND_PROGRESS_DEFAULT_CLASS
-    percent = 100 * elapsed.days // full.days if full.days > 0 else 100
-    context.update(
-        {
-            "current": current_question,
-            "start": start,
-            "end": end,
-            "full": full,
-            "remaining": remaining,
-            "elapsed": elapsed,
-            "percent": percent,
-            "progressbar_class": progressbar_class,
-        }
-    )
+    data = utils.get_progressbar_data(start, end)
+    context.update({"current": current_question, **data})
     return context
 
 
 @register.filter
 def progress_time(delta):
-    if delta.days >= 1:
-        count = delta.days
-        return _("%(count)d day", "%(count)d days", count) % {"count": count}
-    elif delta.seconds // 3600 >= 1:
-        count = delta.seconds // 3600
-        return _("%(count)d hour", "%(count)d hours", count) % {"count": count}
-    elif delta.seconds // 60 >= 1:
-        count = delta.seconds // 60
-        return _("%(count)d minute", "%(count)d minutes", count) % {"count": count}
-    else:
-        count = delta.seconds
-        return _("%(count)d second", "%(count)d seconds", count) % {"count": count}
+    return utils.progress_time(delta)

```

Comment:
Toto asi moze byt spolocny filter, nie?
Teda staci ho registrovat raz, netreba mat zvlast toto aj ten v statements.

Reply:
Ked ten filter zaregistrujem len v polls_parts, potom nebude fungovat v statements, a naopak.
Ako funguje to loadovanie filtrov? Ked napisem {% load XXX %}, tak kde sa hlada to XXX? V polls_parts alebo sa to vie najst aj v utils?

Reply:
Loadovanie filtrov funguje podla module name, detaily: https://docs.djangoproject.com/en/3.0/ref/templates/builtins/#std:templatetag-load

Takze mozes pouzit:
{% load utils %} resp {% load progressbar %} ak to das do progressbar package.